test/e2e_node/util.go

```diff
@@ -60,3 +70,155 @@ func getNodeSummary() (*stats.Summary, error) {
 	}
 	return &summary, nil
 }
+
+// Returns the current KubeletConfiguration
+func getCurrentKubeletConfig() (*componentconfig.KubeletConfiguration, error) {
+	resp := pollConfigz(5*time.Minute, 5*time.Second)
+	kubeCfg, err := decodeConfigz(resp)
+	if err != nil {
+		return nil, err
+	}
+	return kubeCfg, nil
+}
+
+func getCurrentKubeletConfigMap(f *framework.Framework) (*api.ConfigMap, error) {
+	return f.ClientSet.Core().ConfigMaps("kube-system").Get(fmt.Sprintf("kubelet-%s", framework.TestContext.NodeName))
+}
+
+// Creates or updates the configmap for KubeletConfiguration, waits for the Kubelet to restart
+// with the new configuration. Returns an error if the configuration after waiting 40 seconds
+// doesn't match what you attempted to set.
+func setKubeletConfiguration(f *framework.Framework, kubeCfg *componentconfig.KubeletConfiguration) error {
```

Comment:
Can we check the feature gates first and only set kubelet configuration when dynamic kubelet configuration is enabled?